test/index.js

```diff
@@ -2175,6 +2175,164 @@ describe('control', function() {
 
   });
 
+  describe('allPass', function() {
+
+    it('is a binary function', function() {
+      eq(typeof S.allPass, 'function');
+      eq(S.allPass.length, 2);
+    });
+
+    it('type checks its arguments', function() {
+      assert.throws(function() { S.allPass('wrong'); },
+                    errorEq(TypeError,
+                            '‘allPass’ requires a value of type Array ' +
+                            'as its first argument; received "wrong"'));
+    });
+
+    it('returns true when given an empty array of predicates', function() {
+      eq(S.allPass([], 'quiessence'), true);
+    });
+
+    it('returns true when all predicates pass', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                   'quiessence'),
+         true);
+    });
+
+    it('returns false when some predicates fail', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                   'quintessential'),
+         false);
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                   'incandescent'),
+         false);
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                   'fissiparous'),
+         false);
+    });
+
+    it('returns false when all predicates fail', function() {
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    S.test(/c/)],
+                   'empathy'),
+         false);
+    });
+
+    it('short-circuits when one predicate fails', function() {
+      var evaluated = false;
+      var evaluate = function() { evaluated = true; };
+
+      eq(S.allPass([S.test(/q/),
+                    S.test(/u/),
+                    S.test(/i/),
+                    evaluate],
+                   'incandescent'),
+         false);
+      eq(evaluated, false);
+    });
+
+    it('is curried', function() {
+      eq(S.allPass([]).length, 1);
+      eq(S.allPass([S.test(/q/)])('quiessence'), true);
```

Comment:
I think it'd read better if the first argument were `[S.test(/q/)]` in both cases (we follow this convention elsewhere).